app/main.py

```diff
@@ -1,25 +1,78 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
-class IntegerRange:
+class Validator(ABC):
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, obj: object, obj_type: object = None) -> None:
+        return getattr(obj, self.protected_name)
+
+    def __set__(self, obj: object, value: int | float) -> None:
+        self.validate(value)
+        setattr(obj, self.protected_name, value)
+
+    @abstractmethod
+    def validate(self, value: int | float) -> None:
+        pass
+
+
+class IntegerRange(Validator):
+
+    def __init__(self, min_amount: int = None, max_amount: int = None) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def validate(self, value: int | float) -> None:
+        if not isinstance(value, int | float):
+            raise TypeError("Quantity should be int or float.")
+
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(
+                f"Quantity should not be less than {self.min_amount}"
+                f"and greater than {self.max_amount}.")
```

Comment:
I don't think you should create 2 classes, you can write everything in IntegerRange

Reply:
Why I must create all in one class? in prevoius task i also created 2 classes Validator and Number. But this is posible to create other validete for namber and OneOf. I think if we used descriptors in previus task and in this task. I can use it.
I wrote onetime validator and use it in other task?  just add float check.

Reply:
> I don't think you should create 2 classes, you can write everything in IntegerRange

Why I must create all in one class? in prevoius task i also created 2 classes Validator and Number. But this is posible to create other validate for Number and OneOf. I think if we used descriptors in previous task and in this task. I can use it.
I wrote onetime validator and use it in other task? just add "float" check.